src/TxHistory/TxHistoryNavigator.tsx

```diff
@@ -31,27 +32,28 @@ const Stack = createStackNavigator<TxHistoryRoutes>()
 export const TxHistoryNavigator = () => {
   const strings = useStrings()
   const wallet = useSelectedWallet()
+
   const walletName = useWalletName(wallet)
   const transactionInfos = useSelector(transactionsInfoSelector)
-  const tokenBalance = useSelector(tokenBalanceSelector)
+  const defaultTokenId = getDefaultAssetByNetworkId(wallet.networkId).identifier
+  const balance = useBalances(wallet, defaultTokenId)
   const [modalInfoState, setModalInfoState] = React.useState(false)
   const showModalInfo = () => setModalInfoState(true)
   const hideModalInfo = () => setModalInfoState(false)
-  const [selectedTokenIdentifier, setSelectedTokenIdentifier] = React.useState(
-    tokenBalance.getDefaultEntry().identifier,
-  )
+
+  const [selectedTokenIdentifier, setSelectedTokenIdentifier] = React.useState(defaultTokenId)
   const [sendAll, setSendAll] = React.useState(false)
   const [receiver, setReceiver] = React.useState('')
   const [amount, setAmount] = React.useState('')
 
-  // when the selected asset is no longer available
-  const selectedAsset = tokenBalance.values.find(({identifier}) => identifier === selectedTokenIdentifier)
-  if (!selectedAsset) {
-    setSelectedTokenIdentifier(tokenBalance.getDefaultEntry().identifier)
-    setSendAll(false)
-    setReceiver('')
-    setAmount('')
-  }
+  useEffect(() => {
```

Comment:
does it need an effect for it?

Reply:
Infinite loop when the token doesn't exist

Reply:
Finally I have restored the previous logic because now the balance object never is empty. Thanks!